js/views/modals/purchase/Shipping.js

```diff
@@ -12,15 +14,11 @@ export default class extends BaseModal {
     if (!this.model || !(this.model instanceof Listing)) {
       throw new Error('Please provide a listing model');
     }
+    this.shippingOptions = this.createChild(ShippingOptions, {
+      model: this.model,
+    });
 
-    this.listenTo(app.settings.get('shippingAddresses'), 'update',
-      (cl, updateOpts) => {
-        if (updateOpts.changes.added.length ||
-          updateOpts.changes.removed.length) {
-          // update the shipping section with the changed address information
-          // TODO: add shipping code here
-        }
-      });
+    this.listenTo(app.settings.get('shippingAddresses'), 'update', () => this.render());
```

Comment:
If any changes are made to the shippingAddresses collection, this view will re-render. This means the state of the selection is lost, but that is unavoidable without adding some complexity since the user might have deleted the currently selected address, or changed the order of the addresses.